internal/auth-service/auth-service.go

```diff
@@ -0,0 +1,116 @@
+package auth_service
+
+import (
+	"context"
+	"github.com/julienschmidt/httprouter"
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	"github.com/liqotech/liqo/pkg/crdClient"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+	"net/http"
+	"time"
+)
+
+type AuthServiceCtrl struct {
+	namespace      string
+	clientset      kubernetes.Interface
+	saInformer     cache.SharedIndexInformer
+	nodeInformer   cache.SharedIndexInformer
+	secretInformer cache.SharedIndexInformer
+}
+
+func NewAuthServiceCtrl(namespace string, kubeconfigPath string) (*AuthServiceCtrl, error) {
+	config, err := crdClient.NewKubeconfig(kubeconfigPath, &discoveryv1alpha1.GroupVersion)
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(clientset, time.Second*30, informers.WithNamespace(namespace))
+
+	saInformer := informerFactory.Core().V1().ServiceAccounts().Informer()
+	saInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{})
+
+	nodeInformer := informerFactory.Core().V1().Nodes().Informer()
+	nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{})
+
+	secretInformer := informerFactory.Core().V1().Secrets().Informer()
+	secretInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{})
+
```

Comment:
i think i'm missing something here. you have three informers but only for the **Secrets** informer you do register valid _EvendHandlerFunctions_. Which is the function of the remaining informers?

Reply:
The other two are used for caching purposes:
* the ServiceAccount one in `serviceAccont.go` at line 60 (this one, I admit, it's a bit useless and I can remove it)
* the Node one in `kubeconfig.go` at line 24